Load the libraries

In [5]:
import pandas as pd 
import numpy as np 
import matplotlib as plt
import seaborn as sns
from wordcloud import WordCloud

Load the data

In [6]:
data = pd.read_csv("netflix_titles.csv")
data.head()

,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,81145628,Movie,Norm of the North: King Sized Adventure,"Richard Finn, Tim Maltby","Alan Marriott, Andrew Toth, Brian Dobson, Cole...","United States, India, South Korea, China","September 9, 2019",2019,TV-PG,90 min,"Children & Family Movies, Comedies",Before planning an awesome wedding for his gra...
1,80117401,Movie,Jandino: Whatever it Takes,NaN,Jandino Asporaat,United Kingdom,"September 9, 2016",2016,TV-MA,94 min,Stand-Up Comedy,Jandino Asporaat riffs on the challenges of ra...
2,70234439,TV Show,Transformers Prime,NaN,"Peter Cullen, Sumalee Montano, Frank Welker, J...",United States,"September 8, 2018",2013,TV-Y7-FV,1 Season,Kids' TV,"With the help of three human allies, the Autob..."
3,80058654,TV Show,Transformers: Robots in Disguise,NaN,"Will Friedle, Darren Criss, Constance Zimmer, ...",United States,"September 8, 2018",2016,TV-Y7,1 Season,Kids' TV,When a prison ship crash unleashes hundreds of...
4,80125979,Movie,#realityhigh,Fernando Lebrija,"Nesta Cooper, Kate Walsh, John Michael Higgins...",United States,"September 8, 2017",2017,TV-14,99 min,Comedies,When nerdy high schooler Dani finally attracts...


Summary of the data

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6234 entries, 0 to 6233
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       6234 non-null   int64 
 1   type          6234 non-null   object
 2   title         6234 non-null   object
 3   director      4265 non-null   object
 4   cast          5664 non-null   object
 5   country       5758 non-null   object
 6   date_added    6223 non-null   object
 7   release_year  6234 non-null   int64 
 8   rating        6224 non-null   object
 9   duration      6234 non-null   object
 10  listed_in     6234 non-null   object
 11  description   6234 non-null   object
dtypes: int64(2), object(10)
memory usage: 584.6+ KB


Insights: 
- we have 6234 entities. Any column less than that entities, containes missing values
[Director, Cast, Country, Date Added]

- Data type inconsistency
[date_added is an object]

- Number of columns: 12


Data Cleaning and Transformation

In [7]:
data.isnull().sum()

show_id            0
type               0
title              0
director        1969
cast             570
country          476
date_added        11
release_year       0
rating            10
duration           0
listed_in          0
description        0
dtype: int64

Handling Director, Country and Cast's null values

In [12]:
data["director"] = data["director"].fillna("Unknown")
data["cast"] = data['cast'].fillna("Unknown")
data["country"] = data['country'].fillna("Unknown")

Handling Date_Added, duration, rating null values

In [13]:
data.dropna(subset=['date_added', 'rating', 'duration'], inplace = True)

Convert datatype of date_added from object to datetime objects

In [17]:
data["date_added"] =  pd.to_datetime(data["date_added"], format="mixed", dayfirst=False)

New Feature for year_added and month_added

In [18]:
data["Year_added"] = data['date_added'].dt.year
data["Month_added"] = data['date_added'].dt.month

In [20]:
data.columns

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in', 'description',
       'Year_added', 'Month_added'],
      dtype='object')

Checking Missing Values and Data Transformation

In [22]:
print("Missing Values")
print(data.isnull().sum())
print('\nDataType after transformation')
print(data.dtypes)

Missing Values
show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
description     0
Year_added      0
Month_added     0
dtype: int64

DataType after transformation
show_id                  int64
type                    object
title                   object
director                object
cast                    object
country                 object
date_added      datetime64[ns]
release_year             int64
rating                  object
duration                object
listed_in               object
description             object
Year_added               int32
Month_added              int32
dtype: object
